File: api/src/main/java/org/delivery/api/domain/storemenu/converter/StoreMenuConverter.java
package org.delivery.api.domain.storemenu.converter;

import org.delivery.common.annotation.Converter;
import org.delivery.common.error.ErrorCode;
import org.delivery.common.exception.ApiException;
import org.delivery.api.domain.storemenu.controller.model.StoreMenuRegisterRequest;
import org.delivery.api.domain.storemenu.controller.model.StoreMenuResponse;
import org.delivery.db.store.StoreEntity;
import org.delivery.db.storemenu.StoreMenuEntity;

import java.util.List;
import java.util.Optional;
import java.util.stream.Collectors;

@Converter
public class StoreMenuConverter {

    public StoreMenuEntity toEntity(
            StoreMenuRegisterRequest request,
            StoreEntity storeEntity
    ){

        return Optional.ofNullable(request)
                .map(it -> StoreMenuEntity.builder()
                        .store(storeEntity)
                        .name(it.getName())
                        .amount(it.getAmount())
                        .thumbnailUrl(it.getThumbnailUrl())
                        .build())
                .orElseThrow(() -> new ApiException(ErrorCode.NULL_POINT));
    }

    public StoreMenuResponse toResponse(StoreMenuEntity entity){

        return Optional.ofNullable(entity)
                .map(it -> StoreMenuResponse.builder()
                        .id(it.getId())
                        .storeId(it.getStore().getId())
                        .name(it.getName())
                        .amount(it.getAmount())
                        .status(it.getStatus())
                        .thumbnailUrl(it.getThumbnailUrl())
                        .likeCount(it.getLikeCount())
                        .sequence(it.getSequence())
                        .build())
                .orElseThrow(() -> new ApiException(ErrorCode.NULL_POINT));
    }

    public List<StoreMenuResponse> toResponse(
            List<StoreMenuEntity> list
    ){
        return list.stream()
                .map(it->toResponse(it))
                .collect(Collectors.toList());
    }
}

File: api/src/main/kotlin/org/delivery/api/domain/storemenu/converter/StoreMenuConverter.kt
package org.delivery.api.domain.storemenu.converter

import org.delivery.api.domain.storemenu.controller.model.StoreMenuRegisterRequest
import org.delivery.api.domain.storemenu.controller.model.StoreMenuResponse
import org.delivery.common.annotation.Converter
import org.delivery.common.error.ErrorCode
import org.delivery.common.exception.ApiException
import org.delivery.db.store.StoreEntity
import org.delivery.db.storemenu.StoreMenuEntity

@Converter
class StoreMenuConverter {

    fun toEntity(
        request: StoreMenuRegisterRequest?,
        storeEntity: StoreEntity?
    ): StoreMenuEntity {

        request?.let {
            return StoreMenuEntity(
                store = storeEntity,
                name = it.name,
                amount = it.amount,
                thumbnailUrl = it.thumbnailUrl,
            )
        }?: throw ApiException(ErrorCode.NULL_POINT)
    }

    fun toResponse(
        entity: StoreMenuEntity?
    ): StoreMenuResponse{

        entity?.let{
            return StoreMenuResponse(
                id = it.id,
                storeId = it.store?.id,
                name = it.name,
                amount = it.amount,
                status = it.status,
                thumbnailUrl = it.thumbnailUrl,
                likeCount = it.likeCount,
                sequence = it.sequence
            )
        }?: throw ApiException(ErrorCode.NULL_POINT)
    }


    fun toResponse(
        list:List<StoreMenuEntity>
    ):List<StoreMenuResponse>{
        return list.map {
            toResponse(it)
        }.toList()
    }
}
